my_compassion/static/src/js/my2_child_center_weather_time.js

```diff
@@ -0,0 +1,91 @@
+const iconClassMap = {
+    "Sun.svg": "weather-icon-sun",
+    "MoonStar.svg": "weather-icon-moon",
+    "CloudRaining04.svg": "weather-icon-rain",
+    "CloudLightning.svg": "weather-icon-rain",
+    "Waves.svg": "weather-icon-mist",
+    "Wind03.svg": "weather-icon-mist",
+    "CloudBlank02.svg": "weather-icon-cloudy",
+};
+document.addEventListener("DOMContentLoaded", () => {
+    const containerEl = document.querySelector(".center-info-card");
+    const timelinePageEl = document.querySelector(".cd-weather-map-container");
+    const currentTimeEl = document.getElementById("current_time");
+    const currentTemperatureEl = document.getElementById("current_temperature");
+    const childId = timelinePageEl.dataset.childId;
+    const timezone = timelinePageEl.dataset.timezone;
+    const weather_icon_el = document.getElementById("weather_icon");
+
+    // Handles time computation and formating
+    if (currentTimeEl) {
+        //Formatting options for the time to be shown
+        const options = {
+            hour: "2-digit",
+            minute: "2-digit",
+            hour12: true,
+            timeZone: timezone,
+        };
+
+        const updateTime = () => {
+            try {
+                const now = new Date();
+                const centerTimeString = now
+                    .toLocaleTimeString("en-US", options)
+                    .replace("AM", "am")
+                    .replace("PM", "pm");
+                currentTimeEl.textContent = centerTimeString;
+            } catch (error) {
+                console.error("Invalid timezone identifier received from server:", timezone, error);
+                clearInterval(clockInterval);
+            }
+        };
+        updateTime();
+        const clockInterval = setInterval(updateTime, 1000 * 15); // Update every 15 seconds
+    }
+    // Fetch and update the current temperature and weather icon
+    if (currentTemperatureEl && weather_icon_el) {
+        fetch(`/my2/children/${childId}/center-weather`)
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error("Network response was not ok");
+                }
+                return response.json();
+            })
+            .then((data) => {
+                const temperature = data.current_temperature;
+                currentTemperatureEl.textContent = `${temperature}`;
+                const icon_id = data.weather_icon_id;
+                weather_icon_el.src = `/theme_compassion_2025/static/src/img/icons/${icon_id}`;
+                //Attach the corresponding class for the icon to allow CSS styling
+                const iconClass = iconClassMap[icon_id];
+                if (iconClass) {
+                    weather_icon_el.classList.add(iconClass);
+                }
+
+                //Upon successful fetch and update, ensure the container is visible
+                if (containerEl) {
+                    containerEl.style.display = "block";
```

Comment:
```suggestion
                    containerEl.classList.add("--visible");
```
I think it’s cleaner to move this into a CSS class or Bootstrap and then just append the `--visible` flag. In my opinion, it’s more flexible, but I guess it’s up to you.